x/millions/keeper/keeper_deposit.go

```diff
@@ -329,6 +329,26 @@ func (k Keeper) hasPoolDeposit(ctx sdk.Context, address string, poolID uint64) b
 	return iterator.Valid()
 }
 
+// UnsafeUpdateAutoCompoundDeposits raw updates deposit's origin
+// It's heavily unsafe and should only be used by store migrations
+func (k Keeper) UnsafeUpdateAutoCompoundDeposits(ctx sdk.Context, deposit types.Deposit, depositOrigin types.DepositOrigin) (types.Deposit, error) {
+	// Acquire the deposit
+	deposit, err := k.GetPoolDeposit(ctx, deposit.PoolId, deposit.DepositId)
+	if err != nil {
+		return types.Deposit{}, err
+	}
+
+	deposit.DepositOrigin = depositOrigin
+	deposit.UpdatedAtHeight = ctx.BlockHeight()
+	deposit.UpdatedAt = ctx.BlockTime()
+	// Update pool deposits
+	k.setPoolDeposit(ctx, &deposit)
+	// Update account deposits
+	k.setAccountDeposit(ctx, &deposit)
+
+	return deposit, nil
+}
+
```

Comment:
The whole method has an issue.

It takes a deposit entity as parameter, but we do replace the same variable by grabbing again the entity from the parameters deposit id.

Method should take a deposit id as parameter, then grab the entity and apply patches.
Last two comments are useless.